src/components/Common/Environments/Switch.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <span>
+    <li>
+      <a class="btn-flat dropdown-button current-environment right grey-text secondary text-lighten-5 waves-effect waves-light"
+         data-activates='environment-dropdown'>
+          <span class="current-environment-name truncate">
+            {{currentEnvironment.name}}
+          </span>
+
+          <i class="fa fa-caret-down"></i>
+      </a>
+    </li>
+
+    <ul id='environment-dropdown' class='dropdown-content'>
+      <li v-for="(id, env) in environments" class="environment">
+        <div @click="switchEnvironment(id)">
+          <span class="name environment-attribute truncate">{{env.name}}</span>
+          <span class="host environment-attribute">{{env.host}}</span>
+        </div>
+        <i class="edit primary fa fa-pencil" @click.prevent="$dispatch('main-menu::create-env', id)"></i>
+        <i class="delete error fa fa-trash" @click.prevent="$dispatch('main-menu::delete-env', id)"></i>
+      </li>
+      <li class="divider"></li>
+      <li><a href="" @click.prevent="$dispatch('main-menu::create-env')"><i class="fa fa-plus-circle"></i> Create new</a></li>
+    </ul>
+  </span>
+</template>
+
+<script>
+  import { environments, currentEnvironment } from '../../../vuex/modules/common/kuzzle/getters'
+  import { switchEnvironment } from '../../../services/environment'
+  import ModalCreate from './ModalCreate'
+
+  export default {
+    name: 'EnvironmentsSwitch',
+    components: {
+      ModalCreate
+    },
+    vuex: {
+      getters: {
+        environments,
+        currentEnvironment
+      }
+    },
+    methods: {
+      switchEnvironment (id) {
+        switchEnvironment(id)
+          .then(() => {
+            // this.$dispatch('toast', `Switched to ${environments[id].name} environment`, 'info')
```

Comment:
Oh, this was me: I actually wanted to give the user a feedback showing that the environment has switched but the toasters seem to be broken...